airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

```diff
@@ -0,0 +1,111 @@
+from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
+
+
+class SnowflakeToS3Operator(BaseOperator):
+    """
+    UNLOAD a query or a table from Snowflake to S3
+
+    :param schema: reference to a specific schema in snowflake database
+    :type schema: str
+    :param table: reference to a specific table in snowflake database
+    :type table: str
+    :param s3_bucket: reference to a specific S3 bucket where the data will be saved
+    :type s3_bucket: str
+    :param s3_key: reference to a specific S3 key within the previous bucket
+    :type s3_key: str
+    :param file_format: can be either a previous file format created in Snowflake console
+    or hardcoded one like
+        ``type = csv
+         field_delimiter = ','
+         skip_header = 1
+         compression=None``
+    :type file_format: str
+    :param sql: optional parameter to unload a customized query instead an entire table
+    :type sql: str
+    :param snowflake_conn_id: reference to a specific snowflake database
+    :type snowflake_conn_id: str
+    :param unload_options: reference to a list of UNLOAD options (SINGLE, MAX_FILE_SIZE,
+    OVERWRITE etc). Each element of the list has to be a string
+    :type unload_options: list
+    """
+
+    template_fields = (
+        's3_key',
+        's3_bucket',
+        'sql',
+        'table',
+        'snowflake_conn_id',
+    )
+    template_ext = ('.sql',)
+    ui_color = '#ffebb2'
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        s3_bucket: str,
+        s3_key: str,
+        file_format: str,
+        schema: Optional[str] = None,
+        table: Optional[str] = None,
+        sql: Optional[str] = None,
+        snowflake_conn_id: str = "snowflake_default",
+        unload_options: Optional[list] = None,
+        autocommit: bool = True,
+        include_header: bool = False,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+        self.file_format = file_format
+        self.snowflake_conn_id = snowflake_conn_id
+        self.schema = schema
+        self.table = table
+        self.sql = sql
+        self.unload_options = unload_options or []  # type: List
+        self.autocommit = autocommit
+        self.include_header = include_header
+
+    def execute(self, context):
+        snowflake_hook = SnowflakeHook(snowflake_conn_id=self.snowflake_conn_id)
+        unload_options = '\n\t\t\t'.join(self.unload_options)
+
+        if not self.sql:
+            self.sql = self.schema + '.' + self.table
+
+        unload_query = f"""
+                    COPY INTO 's3://{self.s3_bucket}/{self.s3_key}'
+                    FROM ({self.sql})
+                    STORAGE_INTEGRATION = S3
+                    FILE_FORMAT = ({self.file_format})
+                    {unload_options}
+                    HEADER = {self.include_header};
+                    """
```

Comment:
Do we need this `\n\t\t\t` because the `unload_query` is indented? Should we be able to dedent it?

Reply:
To be honest, I just copied it from the existing RedshiftToS3Operator. I don't know if we should maintain this or modifying. What do you think?

Reply:
I think if we can improve the code and remove this strange string then it may be worth doing it

Reply:
I was able to remove two \t characters. Otherwise, the query in the log is unintended.